### ISSUE TITLE
2FA via email

### ISSUE THREAD

**Issue by stephenafamo** [volatiletech/authboss#187]
Are there plans of allowing 2FA via email?

Or can the 2FA-SMS module be modified to be more generic? 
That way, someone can decide to send the code through SMS, Email, Push, Call, e.t.c.

**Comment by aarondl** [comment]
There's no plans for any more 2fa options. But I'd accept PRs to add options.

I never considered that email or call actually has the same flow (never heard of push).

Awkwardly enough you could just use the sms2fa module with email today. As long as you modify the templates to remove the traces of SMS, the routes hardly matter to users. Then you just implement an sms2fa.Sender that sends emails and it seems to me like it would just work?

Thoughts?

**Comment by stephenafamo** [comment]
I agree that the module could be used for Email, Push, Voice or any other method.That's why I suggest that the module be renamed to something more generic. Then the documentation can be updated to explain how the OTP can be sent through diffrent methods.On Sep 16, 2018 5:36 PM, Aaron L <notifications@github.com> wrote:There's no plans for any more 2fa options. But I'd accept PRs to add options.
I never considered that email or call actually has the same flow (never heard of push).
Awkwardly enough you could just use the sms2fa module with email today. As long as you modify the templates to remove the traces of SMS, the routes hardly matter to users. Then you just implement an sms2fa.Sender that sends emails and it seems to me like it would just work?
Thoughts?

**Comment by stephenafamo** [comment]
I totally understand.

When I start implementing this on the project I'm working on(maybe in a couple months), I'll definitely make a PR for this.
